pywb/static/wombatProxyMode.js

```diff
@@ -0,0 +1,359 @@
+            return function (A, B, C, D, E, F, G) {
+                // Apply doesn't work for constructors and Date doesn't
+                // seem to like undefined args, so must explicitly
+                // call constructor for each possible args 0..7
+                if (A === undefined) {
+                    return new Date(orig_now() - timediff);
+                } else if (B === undefined) {
+                    return new Date(A);
+                } else if (C === undefined) {
+                    return new Date(A, B);
+                } else if (D === undefined) {
+                    return new Date(A, B, C);
+                } else if (E === undefined) {
+                    return new Date(A, B, C, D);
+                } else if (F === undefined) {
+                    return new Date(A, B, C, D, E);
+                } else if (G === undefined) {
+                    return new Date(A, B, C, D, E, F);
+                } else {
+                    return new Date(A, B, C, D, E, F, G);
+                }
+            }
+        }($wbwindow.Date);
+
+        $wbwindow.Date.prototype = orig_date.prototype;
+
+        $wbwindow.Date.now = function () {
+            return orig_now() - timediff;
+        };
+
+        $wbwindow.Date.UTC = orig_utc;
+        $wbwindow.Date.parse = orig_parse;
+
+        $wbwindow.Date.__WB_timediff = timediff;
+
+        Object.defineProperty($wbwindow.Date.prototype, "constructor", {value: $wbwindow.Date});
+    }
+
+    //============================================
+    function init_disable_notifications() {
+        if (window.Notification) {
+            window.Notification.requestPermission = function (callback) {
+                if (callback) {
+                    callback("denied");
+                }
+
+                return Promise.resolve("denied");
+            };
+        }
+
+        if (window.geolocation) {
+            var disabled = function (success, error, options) {
+                if (error) {
+                    error({"code": 2, "message": "not available"});
+                }
+            };
+
+            window.geolocation.getCurrentPosition = disabled;
+            window.geolocation.watchPosition = disabled;
+        }
+    }
+
+    function initAutoFetchWorker() {
+        if (!$wbwindow.Worker) {
+            return;
+        }
+
+        var isTop = $wbwindow.self === $wbwindow.top;
+
+        function AutoFetchWorker() {
+            if (!(this instanceof AutoFetchWorker)) {
+                return new AutoFetchWorker();
+            }
+            this.checkIntervalCB = this.checkIntervalCB.bind(this);
+            if (isTop) {
+                // Cannot directly load our worker from the proxy origin into the current origin
+                // however we fetch it from proxy origin and can blob it into the current origin :)
+                var self = this;
+                fetch(wb_info.static_prefix + 'autoFetchWorkerProxyMode.js')
```

Comment:
Small request here: could the prefix be made overridable, eg:
```
var wbAutoFetchWorkerPrefix = (wb_info.auto_fetch_worker_prefix || wb_info.static_prefix) + 'autoFetchWorkerProxyMode.js'
```
In WR, will route it to a different prefix to add cors headers..

Reply:
done